src/main/java/org/broadinstitute/hellbender/engine/HaploidWalker.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright (c) 2016, Daniel Gomez-Sanchez <daniel.gomez.sanchez@hotmail All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without modification, are permitted
+ * provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this list of conditions
+ * and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice, this list of conditions
+ * and the following disclaimer in the documentation and/or other materials provided with the
+ * distribution.
+ *
+ * 3. Neither the name of the copyright holder nor the names of its contributors may be used to endorse
+ * or promote products derived from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND
+ * FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER
+ * IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF
+ * THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ */
+
+package org.broadinstitute.hellbender.engine;
+
+import org.magicdgs.thaplv.cmd.argumentcollections.HaplotypeModelArgumentCollection;
+import org.magicdgs.thaplv.cmd.argumentcollections.HaplotypeModelWithPloidyArgumentCollection;
+import org.magicdgs.thaplv.cmd.argumentcollections.HaplotypeModelWithoutPloidyArgumentCollection;
+import org.magicdgs.thaplv.haplotypes.model.HaplotypeModel;
+import org.magicdgs.thaplv.haplotypes.model.VariantHaplotypeConverter;
+
+import htsjdk.variant.variantcontext.VariantContext;
+import org.broadinstitute.hellbender.cmdline.ArgumentCollection;
+import org.broadinstitute.hellbender.engine.filters.VariantFilter;
+import org.broadinstitute.hellbender.engine.filters.VariantFilterLibrary;
+import org.broadinstitute.hellbender.utils.SimpleInterval;
+
+import java.util.stream.StreamSupport;
+
+/**
+ * Variant walker which transform the {@link VariantContext} into haploid. Thus, the variants
+ * passed to {@link #apply(VariantContext, ReadsContext, ReferenceContext, FeatureContext)} have
+ * haploid individuals.
```

Comment:
Indicate that taking the first allele of each `Genotype` contained in the `VariantContext` should be ok, unless allowsDontCheck is true and the user provide an incorrect haploid VCF.